### PR TITLE
feat: Avoid race condition for notice containers

### PR DESCRIPTION
**Pull request by aababilov**
Notice container is not thread-safe, otherwise it will slow down adding
new notices. Each thread now has its own instance of notice container.
When the thread ends, its container is added to the global one.

### CODE REVIEW

File: core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java [hunk 1]
```diff
@@ -116,10 +119,11 @@ public GtfsFeedContainer loadAndValidate(GtfsInput gtfsInput, GtfsFeedName feedN
         }
         try {
             try {
-                List<Future<GtfsTableContainer>> results = exec.invokeAll(loaderCallables);
-                results.forEach(f -> {
+                exec.invokeAll(loaderCallables).forEach(f -> {
```
Comment (lionel-nj) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java:122]: I think the code would be more clear if `f` was renamed, maybe as `tableAndNoticeContainer` or `container`.

Comment (lionel-nj) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java:122]:
```suggestion
                exec.invokeAll(loaderCallables).forEach(container -> {
```

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks @aababilov! Just one suggestion :)

File: core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java (hunk 1 above)
Comment (aababilov) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java:122]: This "f" is for future. We start from 

```
TableAndNoticeContainers containers = f.get();
```

and do not use it after that.

Comment (aababilov) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java:122]: D'accord!

Comment (lionel-nj) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/table/GtfsFeedLoader.java:122]: Oh okay! I missed that one, thanks.

**Review by lionel-nj — APPROVED** [review]
Thanks @aababilov! Will wait for @barbeau's final approval to merge.

File: core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java [hunk 2]
```diff
@@ -75,4 +75,16 @@ public String exportJson() {
         }
         return noticesByType;
     }
+
+    /**
+     * Adds all notices from another container.
+     * <p>
+     * This is useful for multithreaded validation: each thread has its own notice container which is merged
```
Comment (barbeau) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java:82]: I suggest adding a comment at the top of the class explaining the general lack of thread safety in this class, so later someone doesn't see the word "multithreaded" and start synchronizing everything.

**Review by barbeau — COMMENTED** [review]
@aababilov LGTM, although one comment about adding more Javadoc explanation about lack of thread safety in `NoticeContainer`

File: core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java (hunk 2 above)
Comment (aababilov) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/notice/NoticeContainer.java:87]: Done.

**Review by barbeau — APPROVED** [review]
Thanks @aababilov!

**Pull request merged**